highway.tex

```diff
@@ -24,807 +26,1323 @@
+%
+What is even worse is that such a DAG must be kept in RAM and not on disk.
+%
+The reason for that is that one must build efficient data structures over the DAG for the sake of detecting finality and checking correctness of units (in particular the LNC).
+%
+Placing such data structures on disk instead, would cause a dramatic slowdown. 
 
-%In particular, one could imagine dishonest validators $A,B,C,D,\ldots$ and $\alpha,\beta,\gamma,\delta,\ldots$. We could imagine that each of $A,B,C,D,\ldots$ equivocates and sends a different chain of messages to each of $\alpha,\beta,\gamma,\delta,\ldots$ who then forward them to all of the honest validators. As long as $\alpha,\beta,\gamma,\delta,\ldots$ ignore votes from each other and from the honest validators, they can continue to include these new messages from $A,B,C,D,\ldots$. Using this strategy an attacker with control of $2m$ validators can create $m^2$ chains of messages that must be processed by all of the honest validators.
+A practical solution for this issue is to divide the protocol exucution into {\it eras}.
+%
+In each era, $K=1000$ new blocks are added to the blockchain, and importantly a new instance of Highway is run in every era.
+%
+This means that, for instance, in era $5$, we consider the block $B_{4999}$ at height $4999$, finalized in era $4$, as the genesis unit.
 
-%We could solve this issue by forcing the validators $\alpha,\beta,\gamma,\delta,\ldots$ to include votes from many other validators in their new votes, thus forcing them to eventually acknowledge that $A,B,C,D,\ldots$ are equivocating.
+If we use eras in such a way, then the validators only need to store blocks that were finalized in the previous eras and the DAG for the current era.
+%
+This also helps a lot with equivocation spam attacks, as equivocators caught in a particular era can be banned from the very beginning of the subsequent eras.
+%
+The honest nodes start a new era in the {\it relaxed} mode and thus do not need to send endorsements because of past equivocations anymore.
 
-%\subsubsection{Audibility}
+\subsection{Sending Less Endorsements}
+Recall that the original endorsement strategy that we introduced in Section~\ref{sec:endorse} was very simple: after seeing an equivocation, endorse every unit by non-equivocating validators.
+%
+While it allows for quite simple arguments that liveness is preserved, it also introduces a non-negligible overhead in case when equivocations are detected in a particular era.
+%
 
-%To deal with the first issue, we need to make sure that a validator cannot send too many messages without these messages being received by a decent fraction of the others. For this we introduce a new validity criteria:
+Below we present a refined strategy that still guarantees that no honest validator gets ever stuck because of lack of endorsements of its own, honest units.
```

Comment:
```suggestion
Below we present a refined strategy that still guarantees that no honest validator ever gets stuck because of lack of endorsements of its own, honest units.
```
(But better to consult with a native speaker, so I don't make things worse! :sweat_smile: @xcthulhu)

Reply:
I agree with your suggestion, @afck